### ISSUE TITLE
Be able to run multiple bin QC metrics

### ISSUE THREAD

**Issue by harper357** [nf-core/mag#713]
### Description of feature

I posed this in the slack, but I think it would be great if it was possible to run multiple bin QC metrics (BUSCO, GUNC, CheckM2, etc) and get all of that data in the MultiQC report.

I noticed #707 which adds some new options, but I don't know if it is also allowing for multiple or not.

**Comment by jfy133** [comment]
I think conceptually this makes sense however it's not so trivial as the output of each one goes into downstream process and possibly a custom script (not written by myself), so it might not be so trivial. It might take a bit of time to get to it

**Comment by harper357** [comment]
Personally, even if it is just reported in the MultiQC report and not used for anything downstream I think it would be a nice thing to have.

Once the mentioned PR is merged, I can try to work up a new PR to try it out. 

Or maybe I should try getting work hours to write up a whole benchmarking pipeline so its easier to compare new software. That seems rather ambitious though.

**Comment by jfy133** [comment]
So had a chance to quickly through the code:

- One area where we have the one or other for BUSCO/CheckM is filtering contigs for completeness/contamination for going into GTDBTkl but this could then be replaced by parameter to say 'which' of the two outputs should be used for that filtering 
- CheckM(1) is required for GUNC (busco doesn't work), but that doesn't block a user from also running BUSCO
- I was mistaken regarding the downstream custom script, that should be fine (BIN_SUMMARY), the script seems to allow inserting all

So actually this might be relatively easy to do in fact (conceptually) ino terms of running everthing  :) 

However only BUSCO is in MultiQC, so adding CheckM/GUNC etc. into MultiQC is a whole other task 😬

**Comment by harper357** [comment]
Oh, hmm. Well maybe that is a better use of my time first then. 

Thanks for looking into this though!

**Comment by jfy133** [comment]
> Oh, hmm. Well maybe that is a better use of my time first then. 


Which bit?

**Comment by harper357** [comment]
Sorry, I meant writing some MultiQC plugins for CheckM, CheckM2 and GUNC.

**Comment by jfy133** [comment]
Yes, that would be good! Divide and conquer (at least once I'm not travelling or have sick kids as I now have 😵‍💫🤦🏼‍♀️)

**Comment by harper357** [comment]
Not sure if I should make a new issue or track just things here, but I have made some MultiQC module PRs:
GTDB (https://github.com/MultiQC/MultiQC/pull/2970)
CheckM2 (https://github.com/MultiQC/MultiQC/pull/2978) 

GUNC will be next. 

Should I make one for CheckM or is it not worth it since there is work on updating the pipeline to use CheckM2?

**Comment by jfy133** [comment]
Dayum, that's fast - thank you very  much @harper357 !

I would still include CheckM as we will still support it for the time being, if  you don't mind?

**Comment by harper357** [comment]
CheckM (https://github.com/MultiQC/MultiQC/pull/2990)

**Comment by dialvarezs** [comment]
I can take this one while the MultiQC modules get released. Right now I'm quite familiar with that part of the code.

**Comment by harper357** [comment]
Looking like the above PRs will be part of MultiQC 1.26. 

I ran out of steam and never finished GUNC. I will see if i can start working on it again during the holidays

**Comment by harper357** [comment]
@dialvarezs Did your PR allow this? In the documentation it reads like only one QC can be run, so I don't know if it is just a hidden option or not.

**Comment by dialvarezs** [comment]
Hi @harper357, it's not possible yet. I started on improving the BUSCO related code (#730) but we are still working with @jfy133 to get  that merged. Once that is ready, I will make the multi bin qc running possible.

**Comment by harper357** [comment]
Just wanted to check in on this again. I happen to have some free time over the next few weeks, so I am happy to help push this forward if I can.

**Comment by dialvarezs** [comment]
Hi @harper357, sorry, I haven’t been able to implement this yet, and I have a few other things planned for MAG before I can get on it. If you have time to take this on, please feel free, that would be a big help!

**Comment by jfy133** [comment]
Done in the above PR!
